src/quicktube.js

```diff
@@ -264,27 +345,38 @@ class Quicktube {
 
 }
 
-// This is a requirement of the YouTube Player API for iframe embeds
+// This seems to be a requirement of the YouTube Player API for iframe embeds
 // https://developers.google.com/youtube/iframe_api_reference#Requirements
 window.onYouTubeIframeAPIReady = () => {
     // TODO investigate whether this is a set requirement
 };
 
-const quicktubeInit = () => {
-    // Inject the YouTube API onto the page.
-    if (!window.YT) {
+const insertApiScript = (url, hasBeenCreated) => {
+    if (!hasBeenCreated) {
         const newScriptTag = document.createElement('script');
-        newScriptTag.src = 'https://www.youtube.com/iframe_api';
-
-        const firstDocumentScript = document.getElementsByTagName('script')[0];
-        firstDocumentScript.parentNode.insertBefore(newScriptTag, firstDocumentScript);
+        newScriptTag.src = url;
+        FIRST_SCRIPT_TAG.parentNode.insertBefore(newScriptTag, FIRST_SCRIPT_TAG);
     }
+};
 
+const quicktubeInit = () => {
     const videos = Array.prototype.slice.call(document.querySelectorAll('[data-quicktube]'));
     videos.forEach((video) => {
+        let videoDomain;
+        if (video.hasAttribute('data-quicktube-vimeo')) {
+            // Inject the Vimeo Player API
+            insertApiScript(VIMEO_API, window.Vimeo);
+            videoDomain = VIMEO_EMBED;
+        } else {
+            // Inject the YouTube API
+            insertApiScript(YOUTUBE_API, window.YT);
+            videoDomain = YOUTUBE_EMBED;
+        }
         const videoId = video.getAttribute('data-quicktube');
         const options = JSON.parse(video.getAttribute('data-quicktube-options'));
-        const player = new Quicktube(videoId, options);
+        const videoGUID = guid();
+        video.setAttribute('data-quicktube-quid', videoGUID);
```

Comment:
I have replaced this with a simpler unique player ID generation. The only place I could find it being used (apart from the logs) is `playEl.setAttribute('data-play-guid', this.playerId);`, but this attribute doesn't seem to be used anywhere else so I'm not sure it's even necessary.